app/src/main/java/com/base/firebase_operations/ImageAdapter.kt

```diff
@@ -0,0 +1,32 @@
+package com.base.firebase_operations
+
+import android.content.Context
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import android.widget.ImageView
+import androidx.recyclerview.widget.RecyclerView
+import com.bumptech.glide.Glide
+
```

Comment:
First we created an class named as Image Adapter class which is extending the RecyclerView.Adapter class here it takes two parameters like first is the context which is the context of the activity where the adapter is being used and second is the uri parameter which is the list of images that will be displayed in the recycler view.
Here the recycler view.adapter class is an abstract class that is a base class provided by the android framework which acts as a bridge between data and the recycler view.
Now there is a inner class called ImageViewholder that is extending the reyclerview.ViewHolder class and it holds the reference to all the views present in the recycler view.
Now we have three methods of the ImageAdapter class helping us in displaying data inside the recycler view.
first method is the oncreateViewholder(),it is responsible for inflating layout for each item of the recyclerview 
like if our list has 5 images then the oncreateviewholder would inflate 5 imageviews for holding the 5 images.
Then one important method is the onBindViewHolder() which is used when we need to display a new item or update an existing item present in the recycler view.Now in this method it has the holder parameter which is holding the Ui item obtained from the xml layout which is been inflated and a position parameter which holds the position of the items present in the recycler view.
Now we just used the position parameter to get the position of current url and then used the Glide library to load that url into our image view.